### PR TITLE
feat: added support for GRPC_MESSAGE_SIZE env var 

### PR DESCRIPTION
**Pull request by dpeer6**
…MaxGRPCMessageSize for argo-server

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

File: server/apiserver/argoserver.go [hunk 1]
```diff
@@ -56,10 +58,14 @@ import (
 	"github.com/argoproj/argo-workflows/v3/workflow/hydrator"
 )
 
-const (
-	// MaxGRPCMessageSize contains max grpc message size
-	MaxGRPCMessageSize = 100 * 1024 * 1024
-)
+func getGRPCMessageSize() int {
```
Comment (alexec) [inline server/apiserver/argoserver.go:61]: I'd just use `func init()` to se the variable - panic in invalid value

**Review by alexec — CHANGES_REQUESTED** [review]

File: server/apiserver/argoserver.go (hunk 1 above)
Comment (dpeer6) [reply server/apiserver/argoserver.go:61]: Oh, ok. I was not familiar with init() fun capabilities, which are very nice and useful.
I will fix and update the PR.
Thanks! David

Comment (dpeer6) [reply server/apiserver/argoserver.go]: Hi @alexec,

I updated the PR as requested. Hope now it is better.
Thank you! David

File: server/apiserver/argoserver.go [hunk 2]
```diff
@@ -96,6 +95,18 @@ type ArgoServerOpts struct {
 	AccessControlAllowOrigin string
 }
 
+func init() {
+	var err error
+	MaxGRPCMessageSize = 100 * 1024 * 1024
+        value, exists := os.LookupEnv("GRPC_MESSAGE_SIZE")
```
Comment (alexec) [inline server/apiserver/argoserver.go:101]: please use `env.GetInt`

File: server/apiserver/argoserver.go [hunk 3]
```diff
@@ -96,6 +95,18 @@ type ArgoServerOpts struct {
...
+        if exists {
+		MaxGRPCMessageSize, err = strconv.Atoi(value)
+                if err != nil {
+			panic(err)
```
Comment (alexec) [inline server/apiserver/argoserver.go:105]: please use `log.Fatalf()` which prints more diagnostics

File: server/apiserver/argoserver.go [hunk 4]
```diff
@@ -246,6 +257,7 @@ func (as *argoServer) newGRPCServer(instanceIDService instanceid.Service, offloa
 
 	grpcServer := grpc.NewServer(sOpts...)
 
+	log.Infof("GRPC Server Max Message Size is: %d", MaxGRPCMessageSize)
```
Comment (alexec) [inline server/apiserver/argoserver.go:260]: no need to log here, log in the init function

Comment (alexec) [reply server/apiserver/argoserver.go:260]: also, please use structured logging

**Review by alexec — CHANGES_REQUESTED** [review]

**Comment by alexec** [comment]
looks good, I've suggested a couple of ways to simplify

File: server/apiserver/argoserver.go (hunk 2 above)
Comment (dpeer6) [reply server/apiserver/argoserver.go]: done

File: server/apiserver/argoserver.go (hunk 3 above)
Comment (dpeer6) [reply server/apiserver/argoserver.go]: done

File: server/apiserver/argoserver.go (hunk 4 above)
Comment (dpeer6) [reply server/apiserver/argoserver.go]: ok, followed that as well

**Comment by dpeer6** [comment]
@alexec, I followed your suggestions and updated the PR.
Thank you!

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
LGTM!

**Comment by alexec** [comment]
@dpeer6 I've reverted the merge because it merged to `release-3.1`. That was my mistake. PRs should be based off `master`. Please can you raise a new PR based on `master`?

**Comment by dpeer6** [comment]
@alexec ok I will raise a new PR.

**Comment by agilgur5** [comment]
> @alexec ok I will raise a new PR.

This was completed in #6420

**Pull request merged**